iothub/device/src/Properties.cs

```diff
@@ -0,0 +1,121 @@
+﻿// Copyright (c) Microsoft. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+
+using System;
+using System.Collections;
+using System.Collections.Generic;
+using Microsoft.Azure.Devices.Shared;
+
+namespace Microsoft.Azure.Devices.Client
+{
+    /// <summary>
+    /// A container for properties.
+    /// </summary>
+    public class Properties : IEnumerable<object>
+    {
+        private const string VersionName = "$version";
+        private readonly IDictionary<string, object> _readOnlyProperties = new Dictionary<string, object>();
```

Comment:
rename this to "_reportedPropertyCollection"

Reply:
Done